### PR TITLE
Fixes issue with calculating IOU precision

### PR DESCRIPTION
**Pull request by N3TC4T**
## High Level Overview of Change

Fixes issue with calculating precision for IOU values when value is negative or exponential notation

### Context of Change

If IOU value contains negative value or value is exponential notation and actual precision is as `_MAX_IOU_PRECISION`  this leads to `verify_iou_value` raise exception as calculation precision will return wrong value.

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

**Comment by N3TC4T** [comment]
I realized there is issue in calculating IOU precision values with scientific notation (5.443957975934530E-8), I will try to update this pull request with a fix for this issue as well

**Comment by N3TC4T** [comment]
I'm still not sure if this is the right way as it won't work for value's with fixed points like `2E+2`.

probably this would be covert all the situations:

```
decimal_value = Decimal(value, Context(prec=100))
# fixed value
if(decimal_value == decimal_value.to_integral()):
    return len(decimal_value.quantize(Decimal(1), context=Context(prec=100)).as_tuple().digits)
return len(decimal_value.normalize(Context()).as_tuple().digits)
```

**Comment by ledhed2222** [comment]
woo hoo! thanks for this PR. would you add test cases for scientific notation, as you pointed out? also, looks like this PR is not passing our linter. would you fix whatever it's complaining about? if you set up the `pre-commit` hooks as described in `CONTRIBUTING` you should see exactly what the issue is locally. let me know if you're having any issues setting that up :)

**Review by ledhed2222 — COMMENTED** [review]
woo hoo! thanks for this PR. would you add test cases for scientific notation, as you pointed out? also, looks like this PR is not passing our linter. would you fix whatever it's complaining about? if you set up the pre-commit hooks as described in CONTRIBUTING you should see exactly what the issue is locally. let me know if you're having any issues setting that up :)

**Comment by ledhed2222** [comment]
also, please add a note in `CHANGELOG` :)

**Comment by ledhed2222** [comment]
hey @N3TC4T have you had a chance to take another look at this?

**Comment by N3TC4T** [comment]
Hey Greg, sorry I was a little busy with releasing XUMM, I will try to update this pull request this weekend, thanks.

File: tests/unit/core/binarycodec/fixtures/data/data-driven-tests.json [hunk 1]
```diff
@@ -2881,21 +2881,6 @@
...
-      },
-      "significant_digits": 1,
-      "type_id": 6,
-      "is_native": false,
-      "mantissa": "00038D7EA4C68000",
-      "type": "Amount",
-      "expected_hex": "E7C38D7EA4C6800000000000000000000000000055534400000000000000000000000000000000000000000000000001",
-      "is_negative": false,
-      "exponent": 62
-    },
```
Comment (mvadari) [inline tests/unit/core/binarycodec/fixtures/data/data-driven-tests.json:2898]: Why was this test removed?

Comment (N3TC4T) [reply tests/unit/core/binarycodec/fixtures/data/data-driven-tests.json:2898]: as the calculation the IOU precision method is fixed, `100000000000000000000000000000000000000000000000000000000000000000000000000000` is not a valid IOU amount anymore, so it's failing under this method https://github.com/XRPLF/xrpl-py/blob/f2f80ccfd8385370362cc3344e4627b19995a030/xrpl/core/binarycodec/types/amount.py#L81-L109

**Review by mvadari — APPROVED** [review]
Awesome, thanks!

**Pull request merged**